mslib/mswms/mss_plot_driver.py

```diff
@@ -676,3 +676,212 @@ def plot(self):
+        """
+        if self.dataset is None:
+            return {}
+        data = {}
+
+        timestep = self.times.searchsorted(self.fc_time)
+        logging.debug("loading data for time step %s (%s)", timestep, self.fc_time)
+
+        # Determine the westmost longitude in the cross-section path. Subtract
+        # one gridbox size to obtain "left_longitude".
+        dlon = self.lon_data[1] - self.lon_data[0]
+        left_longitude = self.lons.min() - dlon
+        logging.debug("shifting data grid to gridpoint west of westmost "
+                      "longitude in path: %.2f (path %.2f).",
+                      left_longitude, self.lons.min())
+
+        # Shift the longitude field such that the data is in the range
+        # left_longitude .. left_longitude+360.
+        # NOTE: This does not overwrite self.lon_data (which is required
+        # in its original form in case other data is loaded while this
+        # file is open).
+        lon_data = ((self.lon_data - left_longitude) % 360) + left_longitude
+        lon_indices = lon_data.argsort()
+        lon_data = lon_data[lon_indices]
+        factors = []
+
+        # Make sure air_pressure is the first to be evaluated
+        variables = list(self.data_vars)
+        if variables[0] != "air_pressure":
+            variables.insert(0, variables.pop(variables.index("air_pressure")))
+
+        for name in variables:
+            var = self.data_vars[name]
+            data[name] = []
+            if len(var.shape) == 4:
+                var_data = var[timestep, ::-self.vert_order, ::self.lat_order, :]
+            else:
+                var_data = var[:][timestep, np.newaxis, ::self.lat_order, :]
+            logging.debug("\tLoaded %.2f Mbytes from data field <%s> at timestep %s.",
+                          var_data.nbytes / 1048576., name, timestep)
+            logging.debug("\tVertical dimension direction is %s.",
+                          "up" if self.vert_order == 1 else "down")
+            logging.debug("\tInterpolating to cross-section path.")
+            # Re-arange longitude dimension in the data field.
+            var_data = var_data[:, :, lon_indices]
+
+            cross_section = utils.interpolate_vertsec(var_data, self.lat_data, lon_data, self.lats, self.lons)
+            # Create vertical interpolation factors and indices for subsequent variables
+            if name == "air_pressure":
+                for index, alt in enumerate(self.alts):
+                    pressures = cross_section[:, index]
+                    closest = 0
+                    direction = 1
+                    for i, pressure in enumerate(pressures):
+                        if abs(pressure - alt) < abs(pressures[closest] - alt):
+                            closest = i
+                            direction = 1 if pressure - alt > 0 else -1
+                    next_closest = closest + direction if closest < len(pressures) - 1 else closest
```

Comment:
what about `closest == 0` and `direction = -1'?

Reply:
Handled now